morpho5/utils/error.c

```diff
@@ -108,6 +108,21 @@ void morpho_writeerrorwithid(error *err, errorid id, int line, int posn, ...) {
     va_end(args);
 }
 
+/** @brief Writes a user error to an error structure
+ *  @param err  The error structure
+ *  @param id   The error id.
+ *  @param message Additional parameters (the data for the printf commands in the message) */
+void morpho_writeusererror(error *err, errorid id, char *message) {
+    err->line=ERROR_POSNUNIDENTIFIABLE;
+    err->posn=ERROR_POSNUNIDENTIFIABLE;
```

Comment:
Why can we not get this information of where the error is thrown from?

Reply:
The error structure currently doesn't track the current instruction. This is actually rather a shame, because it would be great to alert the user where a runtime error occurred. The line/posn can be pulled by the cli from the debug info generated by the compiler...